include/gz/rendering/Material.hh

```diff
@@ -220,8 +221,16 @@ namespace gz
       public: virtual std::string Texture() const = 0;
 
       /// \brief Set the material texture
-      /// \param[in] _texture URI of the new texture file
-      public: virtual void SetTexture(const std::string &_texture) = 0;
+      /// \param[in] _texture URI of the new texture file or identifying name
+      /// if _img is set
+      /// \param[in] _img Image data
+      public: virtual void SetTexture(const std::string &_texture,
+        const std::shared_ptr<common::Image> &_img = nullptr) = 0;
+
+      /// \brief Get the texture data
+      /// \return Pointer to the common::Image with the data if the texture
+      /// was loaded from memory
+      public: virtual std::shared_ptr<common::Image> TextureData() const = 0;
```

Comment:
Same question as the `gz-common` PR about why this is a shared pointer, and whether the image should be const.